services/imageproxy/atmos_camo.go

```diff
@@ -45,22 +54,32 @@ func (backend *AtmosCamoBackend) GetImageDirect(imageURL string) (io.ReadCloser,
 
 func (backend *AtmosCamoBackend) getAtmosCamoImageURL(imageURL string) string {
 	cfg := *backend.proxy.ConfigService.Config()
-	siteURL := *cfg.ServiceSettings.SiteURL
-	proxyURL := *cfg.ImageProxySettings.RemoteImageProxyURL
 	options := *cfg.ImageProxySettings.RemoteImageProxyOptions
 
-	return getAtmosCamoImageURL(imageURL, siteURL, proxyURL, options)
-}
+	if imageURL == "" || backend.siteURL == nil {
+		return imageURL
+	}
+
+	// Parse url, bypass if error in parsing.
+	parsedURL, err := url.Parse(imageURL)
+	if err != nil {
+		return imageURL
```

Comment:
We should not return the original URL if it can't be parsed. Browsers are less strict and will likely still parse it, allowing more proxy bypasses.

Reply:
What do you want me to return in that place?

Reply:
What do you want me to return in its place?

Reply:
Not sure. Maybe just the siteURL or some other static dummy URL? 0/5 on whether that makes sense.

Reply:
Gotcha, will return the siteURL then.